### ISSUE TITLE
<gmpx-place-picker>'s overlay doesn't work with <dialog> or [popover]

### ISSUE THREAD

**Issue by KeithHenry** [googlemaps/extended-component-library#203]
When `<gmpx-place-picker>` loads it appends `<div class="pac-container">` to end of the `<body>` and then absolutely positions it as the overlay to show completions.

In a `<dialog>` or `<... popover="auto|manual">` this fails because the `div.pac-container` is behind the new top layer content. 

The overlay should be inside the shadow DOM for `<gmpx-place-picker>`, and possibly use `popover` itself to force top layer and break out of `overflow: hidden`.

To reproduce:

```html
<gmpx-api-loader key="YOUR-KEY"></gmpx-api-loader>
other content on page
<dialog>
  This address picker does not work:
  
  <gmpx-place-picker></gmpx-place-picker>
</dialog>
```
```css
::backdrop {
  backdrop-filter: blur(3px) brightness(60%); 
}
```
```js
const dialog = document.querySelector("dialog");
dialog.showModal();
```

**Comment by wangela** [comment]
If you would like to upvote the priority of this issue, please comment below or react on the original post above with :+1: so we can see what is popular when we triage.
--------

@KeithHenry Thank you for opening this issue. 🙏
Please check out these other resources that might help you get to a resolution in the meantime:

- [Check the issue tracker](https://developers.google.com/maps/support/#issue_tracker) - bugs and feature requests for Google Maps Platform APIs and SDKs
- [Open a support case](https://developers.google.com/maps/support#contact-maps-support) - Get 1:1 support in Cloud Console.
- [Discord](https://discord.gg/fHTDmZe9En) - chat with other developers
- [StackOverflow](http://stackoverflow.com/questions/tagged/google-maps) - use the `google-maps` tag


_This is an automated message, feel free to ignore._

**Comment by KeithHenry** [comment]
@wangela thanks, but this is specifically a bug in the implementation of the overlay - it should be fixable in a PR here

**Comment by KeithHenry** [comment]
The workaround for this is to use a mutation observer, something like:

```ts
  -- in a web component wrapping <gmpx-place-picker>

    #pacContainerObserver = new MutationObserver((mutationsList, observer) => this.hoistPacToContext(mutationsList, observer))

    private hoistPacToContext(mutations: MutationRecord[], observer: MutationObserver) {
        // Find the overlay in the body and move it this component
        for (const mutation of mutations)
            for (const node of mutation.addedNodes) 
                if (node instanceof HTMLElement && node.classList.contains('pac-container'))
                    this.shadowRoot.appendChild(node);

    }

    connectedCallback() {
        super.connectedCallback();
        this.#pacContainerObserver.observe(document.body, { childList: true, subtree: false });
    }

    disconnectedCallback() {
        super.disconnectedCallback();
        this.#pacContainerObserver.disconnect();
    }
```

Then override the styles of that `<div class="pac-container">` as the styles the API injects are in the head not the shadow DOM, and you need to handle other stacking contexts or `position: relative`.

The problem with this hack is that you can only have one address lookup in the page at a time, as it will 'steal' any `.pac-container` created by any others.

**Comment by awmack** [comment]
Thanks for raising, and the suggested workaround. As you discovered, this appears to have the same root cause as another issue, so I'll consolidate the two.

**Comment by awmack** [comment]
Duplicate of #186

**Comment by KeithHenry** [comment]
> Thanks for raising, and the suggested workaround. As you discovered, this appears to have the same root cause as another issue, so I'll consolidate the two.

@awmack OK and fair enough, but I'd add that while these issues have the same cause, it would be possible to fix that one with additional positioning JS, while this can only be fixed by moving the overlay elements into the Shadow DOM.

It's possible to fix that without fixing this.

**Comment by awmack** [comment]
Thanks for pointing that out! The fix will involve switching to the new [Place Autocomplete](https://developers.google.com/maps/documentation/javascript/place-autocomplete-new) in the Maps JS API, which does exactly what you're proposing (so we will indeed fix #203 and #186 simultaneously).
